### PR TITLE
Avoid doing string formatting when calling checkArgument.

### PR DESCRIPTION
**Pull request by bogdandrutu**
This improves performance significantly for our propagations:
Before (mean latency):
propagation.B3FormatImplBenchmark.extract  sample  163623       1993.963 ±   61.193  ns/op

After (mean latency):
propagation.B3FormatImplBenchmark.extract  sample  142707        174.384 ±    26.763  ns/op

### CODE REVIEW

File: api/src/main/java/io/opencensus/metrics/Metric.java [hunk 1]
```diff
@@ -81,19 +81,16 @@ private static void checkTypeMatch(MetricDescriptor.Type type, List<TimeSeries>
           case GAUGE_INT64:
           case CUMULATIVE_INT64:
             Utils.checkArgument(
-                value instanceof ValueLong,
-                String.format("Type mismatch: %s, %s.", type, valueClassName));
+                value instanceof ValueLong, "Type mismatch: %s, %s.", type, valueClassName);
```
Comment (songy23) [inline api/src/main/java/io/opencensus/metrics/Metric.java]: Do you want to split this change to another commit? (`metrics` is not present in the v0.15 release)

Comment (bogdandrutu) [reply api/src/main/java/io/opencensus/metrics/Metric.java]: Done.
